src/wp-admin/includes/block-editor.php

```diff
@@ -0,0 +1,241 @@
+	$image_size_names = apply_filters(
+		'image_size_names_choose',
+		array(
+			'thumbnail' => __( 'Thumbnail' ),
+			'medium'    => __( 'Medium' ),
+			'large'     => __( 'Large' ),
+			'full'      => __( 'Full Size' ),
+		)
+	);
+
+	$available_image_sizes = array();
+	foreach ( $image_size_names as $image_size_slug => $image_size_name ) {
+		$available_image_sizes[] = array(
+			'slug' => $image_size_slug,
+			'name' => $image_size_name,
+		);
+	}
+
+	$default_size       = get_option( 'image_default_size', 'large' );
+	$image_default_size = in_array( $default_size, array_keys( $image_size_names ), true ) ? $image_default_size : 'large';
+
+	$image_dimensions = array();
+	$all_sizes        = wp_get_registered_image_subsizes();
+	foreach ( $available_image_sizes as $size ) {
+		$key = $size['slug'];
+		if ( isset( $all_sizes[ $key ] ) ) {
+			$image_dimensions[ $key ] = $all_sizes[ $key ];
+		}
+	}
+
+	$editor_settings = array(
+		'alignWide'              => get_theme_support( 'align-wide' ),
+		'allowedBlockTypes'      => $allowed_block_types,
+		'allowedMimeTypes'       => get_allowed_mime_types(),
+		'disableCustomColors'    => get_theme_support( 'disable-custom-colors' ),
+		'disableCustomFontSizes' => get_theme_support( 'disable-custom-font-sizes' ),
+		'disableCustomGradients' => get_theme_support( 'disable-custom-gradients' ),
+		'enableCustomLineHeight' => get_theme_support( 'custom-line-height' ),
+		'enableCustomSpacing'    => get_theme_support( 'custom-spacing' ),
+		'enableCustomUnits'      => get_theme_support( 'custom-units' ),
+		'isRTL'                  => is_rtl(),
+		'imageDefaultSize'       => $image_default_size,
+		'imageDimensions'        => $image_dimensions,
+		'imageEditing'           => true,
+		'imageSizes'             => $available_image_sizes,
+		'maxUploadFileSize'      => $max_upload_size,
+	);
+
+	// Theme settings.
+	$color_palette    = current( (array) get_theme_support( 'editor-color-palette' ) );
+	$font_sizes       = current( (array) get_theme_support( 'editor-font-sizes' ) );
+	$gradient_presets = current( (array) get_theme_support( 'editor-gradient-presets' ) );
+
+	if ( false !== $color_palette ) {
+		$editor_settings['colors'] = $color_palette;
+	}
+
+	if ( false !== $font_sizes ) {
+		$editor_settings['fontSizes'] = $font_sizes;
+	}
+
+	if ( false !== $gradient_presets ) {
+		$editor_settings['gradients'] = $gradient_presets;
+	}
+
+	return $editor_settings;
+}
+
+/**
+ * Initialize a block-based editor.
+ *
+ * @since 5.8.0
+ *
+ * @param string $script_name     Name of the script that initializes the editor.
+ * @param array  $editor_settings Optional editor settings.
+ * @param array  $other_settings {
+ *     Other settings to use when initializing the block-based editor.
+ *
+ *     @type array $preload_paths Optional list of paths to preload.
+ * }
+ *
+ * @return void
+ */
+function block_editor_initialize( $script_name, $editor_settings, $other_settings ) {
```

Comment:
While I think there's value in the function above, I'm not sure there's value in this one. I think initialization can be too deferrent between the screens and this feels like a function that is there to "factorize" things but without semantics

Reply:
How about we strip out the initialization part for the `wp-edit-post` with all editor settings? Instead, we rename it to `block_editor_configure`, and leave the parts responsible for configuring preload paths, block categories, block definitions, and so on?

Reply:
yeah that's better.

Reply:
I updated PR to have `block_editor_configure`.